src/main/java/com/upsolver/datasources/jdbc/JDBCDataSource.java

```diff
@@ -303,10 +306,19 @@ private ResultSet queryData(JDBCTaskMetadata metadata, int limit, Connection con
                     timestampCols,
                     fullLoad);
         } catch (SQLException e) {
-            return Collections.singletonList(new PropertyError(connectionStringProp, "Unable to connect to database, please ensure connection string and login info is correct.\n" +
-                    "SqlError: " + e.getMessage()));
+            final String errorMessage;
+            if (sqlDrivers.getDrivers().stream().filter(driver -> connectionString.startsWith(driver.getUrlPrefix())).findAny().isEmpty()) {
+                logger.warn("Unable to connect to database, using JDBC URL: {}", connectionString, e);
+                String msg = sqlDrivers.getDrivers().stream().map(driver -> format("%s (%s)",
+                        driver.getName(),
+                        driver.getUrlTemplate().replace("[", "\\[").replace("]", "\\]")))
```

Comment:
use a more generic markdown escape method + at the very least comment that this is for markdown escaping.

Reply:
Created utility that implements this trivial escaping method. The utility may be improved in future if needed.